ClosedXML/Excel/CalcEngine/Functions/Information.cs

```diff
@@ -27,95 +27,136 @@ public static void Register(CalcEngine ce)
             ce.RegisterFunction("TYPE", 1, Type);
         }
 
-        static IDictionary<ErrorExpression.ExpressionErrorType, int> errorTypes = new Dictionary<ErrorExpression.ExpressionErrorType, int>()
+        private static object ErrorType(List<Expression> p)
         {
-            [ErrorExpression.ExpressionErrorType.NullValue] = 1,
-            [ErrorExpression.ExpressionErrorType.DivisionByZero] = 2,
-            [ErrorExpression.ExpressionErrorType.CellValue] = 3,
-            [ErrorExpression.ExpressionErrorType.CellReference] = 4,
-            [ErrorExpression.ExpressionErrorType.NameNotRecognized] = 5,
-            [ErrorExpression.ExpressionErrorType.NumberInvalid] = 6,
-            [ErrorExpression.ExpressionErrorType.NoValueAvailable] = 7
-        };
-
-        static object ErrorType(List<Expression> p)
-        {
-            var v = p[0].Evaluate();
+            try
+            {
+                p[0].Evaluate();
+            }
+            catch (NullValueException)
+            {
+                return 1;
+            }
+            catch (DivisionByZeroException)
+            {
+                return 2;
+            }
+            catch (CellValueException)
+            {
+                return 3;
+            }
+            catch (CellReferenceException)
+            {
+                return 4;
+            }
+            catch (NameNotRecognizedException)
+            {
+                return 5;
+            }
+            catch (NumberException)
+            {
+                return 6;
+            }
+            catch (NoValueAvailableException)
+            {
+                return 7;
+            }
```

Comment:
Not insisting, but this can be shortened:
```suggestion
            catch (CalcEngineException ex)
            {
                return ex switch
                {
                    NullValueException _ => 1,
                    DivisionByZeroException _ => 2,
                    CellValueException _ => 3,
                    CellReferenceException _ => 4,
                    NameNotRecognizedException _ => 5,
                    NumberException _ => 6,
                    NoValueAvailableException _ => 7,
                    _ => throw new NoValueAvailableException()
                };
            }
```

Reply:
Thanks, I didn't realize that. The shorter version is far more readable.